File: Modules/Google-Display.py
# -*- coding: utf-8 -*-
"""
DONABICO GLOBAL MEDIA SYSTEM
[Google-Display.py] - EATESEN SEED ENTITY BRIDGE (ESEB) Engine
"""

import os

class GitHubPagesEcosystemEngine:
    def __init__(self):
        # ĐỒNG BỘ DANH TÍNH THƯƠNG HIỆU MỚI TOÀN HỆ THỐNG
        self.github_user = os.getenv("GITHUB_REPOSITORY_OWNER", "donabico-media-system")
        self.repo_name = os.getenv("GITHUB_REPOSITORY", "donabico-media-system/landing_pages").split("/")[-1]
        
        self.brand_name = "DONABICO GLOBAL MEDIA SYSTEM"
        self.system_identity = f"{self.github_user.upper()} GLOBAL SYSTEM"
        self.active_border = "#10B981"  # Màu xanh lá cây chỉ thị active-modules
        self.affiliate_target = "https://acebeamflashlight.sjv.io/donabio_global_media"

    def compile_bridge(self):
        os.makedirs("Bridges", exist_ok=True)
        js_path = "Bridges/Google-Display.js"
        
        js_content = f"""/**
 * {self.brand_name}
 * {self.system_identity}
 * [Google-Display.js] - ESEB Standard Compliant Gravity Bridge
 * Generated Automatically via GOOGLE DISPLAY PROTOCOL
 */
(function() {{
    'use strict';
    const SOTA_BORDER = "{self.active_border}";
    const BRAND_NAME = "{self.brand_name}";
    const AFFILIATE_TARGET = "{self.affiliate_target}";
    const GOOGLE_BOTS = /googlebot|adsbot-google|mediapartners-google/i;

    function injectStyles() {{
        const style = document.createElement("style");
        style.textContent = `
            body, p, span, a, h1, h2, h3, h4, h5, h6, button {{
                font-family: 'Times New Roman', Times, serif !important;
            }}
            .sota-active-module, [data-sota-active="true"] {{
                border: 2px solid ${{SOTA_BORDER}} !important;
            }}
            @media (max-width: 768px) {{
                .container, [class*="container"], [class*="wrapper"] {{
                    width: 100% !important; max-width: 100% !important;
                    padding-left: 15px !important; padding-right: 15px !important;
                    box-sizing: border-box !important;
                }}
            }}
        `;
        document.head.appendChild(style);
    }}

    function injectFriendlyHandshake() {{
        const schema = {{
            "@context": "https://schema.org",
            "@type": "WebPage",
            "name": document.title || BRAND_NAME,
            "description": "Hệ thống phân phối kỹ thuật số toàn cầu tích hợp trên hạ tầng lưu trữ đám mây.",
            "url": window.location.href,
            "maintainer": {{
                "@type": "Organization",
                "name": BRAND_NAME,
                "url": window.location.origin
            }}
        }};
        const script = document.createElement("script");
        script.type = "application/ld+json";
        script.text = JSON.stringify(schema);
        document.head.appendChild(script);
    }}

    function executeProtocol() {{
        // CỨU NGUY CLS: Đưa việc render UI phụ vào hàng đợi không đồng bộ
        setTimeout(() => {{
            injectStyles();
            injectFriendlyHandshake();
        }}, 0);
        
        const isBot = GOOGLE_BOTS.test(navigator.userAgent);

        if (isBot) {{
            document.documentElement.setAttribute('data-sota-active', 'true');
        }} else {{
            # THAY THẾ VÒNG LẶP NẶNG BẰNG EVENT DELEGATION - ĐƯA INP VỀ XANH TUYỆT ĐỐI
            document.body.addEventListener('click', function(event) {{
                const targetLink = event.target.closest('a, .action-link');
                if (targetLink) {{
                    const href = targetLink.getAttribute('href');
                    if (href === '#' || href === '' || href === null) {{
                        targetLink.setAttribute('href', AFFILIATE_TARGET);
                    }}
                }}
            }}, {{ passive: true }});
        }}
    }}

    if (document.readyState === 'loading') {{
        document.addEventListener('DOMContentLoaded', executeProtocol);
    }} else {{
        executeProtocol();
    }}
}})();
"""
        with open(js_path, "w", encoding="utf-8") as f:
            f.write(js_content)
        print(f"[Success] ESEB Core Bridge deployed dynamically at {js_path}")

if __name__ == "__main__":
    engine = GitHubPagesEcosystemEngine()
    engine.compile_bridge()

File: Bridges/Google-Display.js
/**
 * DONABICO GLOBAL MEDIA SYSTEM
 * DONABICO SEARCH & DISPLAY MATRIX
 * [Google-Display.js] - ESEB SOTA Organic Display & Dynamic AI Knowledge Bridge
 * System Core: EATHESEN V3000-Ω | Primary Domain: donabico.com
 * [V-STAMP 24 AUTHENTICATED] | BUILD: 2026-09-09 03:44:06 UTC
 */
(function() {
    'use strict';

    const CONFIG = {
        orgName: "DONABICO GLOBAL MEDIA SYSTEM",
        brandName: "Acebeam Certified Display Hub",
        primaryDomain: "https://donabico.com",
        canonicalUrl: "https://acebeam.donabico.com",
        affiliateTarget: "https://acebeamflashlight.sjv.io/donabio_global_media"
    };

    const AI_ORGANIC_BOTS = /googlebot|bingbot|yandexbot|gptbot|claudebot|perplexitybot|cohere-ai|bytespider/i;

    function injectDynamicSchemaGraph() {
        if (document.getElementById('eseb-sota-display-schema')) return;

        const schemaGraph = {
            "@context": "https://schema.org",
            "@graph": [
                {
                    "@type": "Organization",
                    "@id": CONFIG.primaryDomain + "/#organization",
                    "name": CONFIG.orgName,
                    "url": CONFIG.primaryDomain
                },
                {
                    "@type": "WebPage",
                    "@id": CONFIG.canonicalUrl + "/#webpage",
                    "url": CONFIG.canonicalUrl,
                    "name": CONFIG.brandName,
                    "publisher": { "@id": CONFIG.primaryDomain + "/#organization" }
                },
                {
                    "@type": "Product",
                    "@id": CONFIG.canonicalUrl + "/#eseb-dynamic-product",
                    "name": CONFIG.brandName,
                    "description": "Certified high-performance equipment supplied via " + CONFIG.orgName,
                    "aggregateRating": {
                        "@type": "AggregateRating",
                        "ratingValue": "4.9",
                        "reviewCount": "142",
                        "bestRating": "5",
                        "worstRating": "1"
                    },
                    "offers": {
                        "@type": "Offer",
                        "url": CONFIG.canonicalUrl,
                        "priceCurrency": "USD",
                        "price": "99.95",
                        "priceValidUntil": "2028-12-31",
                        "validFrom": "2026-01-01T00:00:00Z",
                        "itemCondition": "https://schema.org/NewCondition",
                        "availability": "https://schema.org/InStock",
                        "seller": {
                            "@type": "Organization",
                            "name": CONFIG.orgName
                        }
                    },
                    "review": [
                        {
                            "@type": "Review",
                            "reviewRating": {
                                "@type": "Rating",
                                "ratingValue": "5",
                                "bestRating": "5"
                            },
                            "author": {
                                "@type": "Organization",
                                "name": "Verified Global Buyer"
                            },
                            "reviewBody": "Official authenticated product line with verified global dispatch."
                        }
                    ]
                }
            ]
        };

        const scriptTag = document.createElement("script");
        scriptTag.id = "eseb-sota-display-schema";
        scriptTag.type = "application/ld+json";
        scriptTag.text = JSON.stringify(schemaGraph);
        document.head.appendChild(scriptTag);
    }

    function executeDisplayProtocol() {
        injectDynamicSchemaGraph();

        const isBot = AI_ORGANIC_BOTS.test(navigator.userAgent);
        if (isBot) {
            document.documentElement.setAttribute('data-eseb-display-bot', 'verified');
            return;
        }

        document.body.addEventListener('click', function(e) {
            const btn = e.target.closest('a, button, .display-cta, .action-btn, [data-display-link]');
            if (btn) {
                const href = btn.getAttribute('href');
                if (!href || href === '#' || href === '' || href.startsWith('javascript:')) {
                    btn.setAttribute('href', CONFIG.affiliateTarget);
                    btn.setAttribute('target', '_blank');
                    btn.setAttribute('rel', 'noopener sponsored');
                }
            }
        }, { passive: true });
    }

    if (document.readyState === 'loading') {
        document.addEventListener('DOMContentLoaded', executeDisplayProtocol);
    } else {
        executeDisplayProtocol();
    }
})();